packages/changed-elements-react/src/widgets/comparisonJobWidget/components/VersionCompareSelectModal.tsx

```diff
@@ -0,0 +1,370 @@
+      targetVersion: args.targetVersion,
+      comparisonJobClient: args.comparisonJobClient,
+      iModelConnection: args.iModelConnection,
+      currentVersion: currentVersion,
+      getDialogOpen: args.getDialogOpen,
+    }).catch((e) => {
+      Logger.logError(VersionCompare.logCategory, "Could not start version comparison: " + e);
+    });
+  }
+};
+
+type RunStartComparisonV2Args = {
+  targetVersion: NamedVersion;
+  comparisonJobClient: IComparisonJobClient;
+  iModelConnection: IModelConnection;
+  currentVersion: NamedVersion;
+  getDialogOpen: () => boolean;
+};
+
+// todo clean this up once API fix is in. API is currently publishing jobs that are complete but mau not have href
+// while loop may need to be looked at again
+const runStartComparisonV2 = async (args: RunStartComparisonV2Args) => {
+  let { comparisonJob } = await postOrGetComparisonJob({
+    changedElementsClient: args.comparisonJobClient,
+    iTwinId: args.iModelConnection?.iTwinId as string,
+    iModelId: args.iModelConnection?.iModelId as string,
+    startChangesetId: args.targetVersion.changesetId as string,
+    endChangesetId: args.currentVersion.changesetId as string,
+  });
+  if (comparisonJob.status === "Completed" && comparisonJob.comparison && comparisonJob.comparison.href) {
+    void runManagerStartComparisonV2({
+      comparisonJob: { comparisonJob: comparisonJob },
+      comparisonJobClient: args.comparisonJobClient,
+      iModelConnection: args.iModelConnection,
+      targetVersion: args.targetVersion,
+      currentVersion: args.currentVersion,
+    });
+    return;
+  }
+  if (comparisonJob.status !== "Completed" && comparisonJob.status !== "Error") {
+    toastComparisonJobProcessing(args.currentVersion, args.targetVersion);
+  }
+  while (comparisonJob.status !== "Error") {
+    await new Promise((resolve) => setTimeout(resolve, 5000)); // run loop every 5 seconds
+    if (VersionCompare.manager?.isComparing) {
+      return;
+    }
+    comparisonJob = (await args.comparisonJobClient.getComparisonJob({
+      iTwinId: args.iModelConnection?.iTwinId as string,
+      iModelId: args.iModelConnection?.iModelId as string,
+      jobId: `${args.targetVersion.changesetId}-${args.currentVersion.changesetId}`,
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })).comparisonJob;
+
+    if (comparisonJob.status === "Completed" && comparisonJob.comparison && comparisonJob.comparison.href) {
+      if (!args.getDialogOpen() && !VersionCompare.manager?.isComparing) {
+        toastComparisonJobComplete({
+          comparisonJob: { comparisonJob: comparisonJob },
+          comparisonJobClient: args.comparisonJobClient,
+          iModelConnection: args.iModelConnection,
+          targetVersion: args.targetVersion,
+          currentVersion: args.currentVersion,
+        });
+      }
+      return;
+    }
+    if (comparisonJob.status === "Error") {
+      toastComparisonJobError(args.currentVersion, args.targetVersion)
```

Comment:
There is still no notification about error when the initial `postOrGetComparisonJob` fails.